src/sentry/tasks/reports.py

```diff
@@ -374,6 +374,10 @@ def get_calendar_query_range(interval, months):
 def clean_calendar_data(project, series, start, stop, rollup, timestamp=None):
     earliest = tsdb.get_earliest_timestamp(rollup, timestamp=timestamp)
 
+    # XXX: This is temporary for sentry.io until 30 days after deployment and
+    # any preexisting records have their TTLs updated.
+    earliest = earliest + (30 * rollup)
```

Comment:
Changed my mind and just made this a setting in TSDB to avoid weird configuration-in-code changes like this that are only pertinent to our installation. If the release would have went out with this hard-coded 30 day shift in here it would have led to on-premise 3 month graphs that only showed 30 days of data which would be pretty stupid.